src/modules/Dashboard/features/Applet/Overview/Overview.tsx

```diff
@@ -1,18 +1,86 @@
+import { useTranslation } from 'react-i18next';
 
+import { DashboardTable } from 'modules/Dashboard/components';
+import { AddParticipantPopup } from 'modules/Dashboard/features/Applet/Popups';
+import { QuickStats } from 'modules/Dashboard/features/Applet/Overview/QuickStats';
 import { Spinner } from 'shared/components';
-import { StyledBody } from 'shared/styles';
+import { StyledFlexColumn, StyledTitleLarge } from 'shared/styles';
+import { GetAppletSubmissionsResponse, getAppletSubmissionsApi } from 'api';
+import { useAsync } from 'shared/hooks';
+import { DEFAULT_ROWS_PER_PAGE } from 'shared/consts';
+
+import { mapResponseToQuickStatProps, mapResponseToSubmissionsTableProps } from './Overview.utils';
+import { StyledRoot } from './Overview.styles';
+
+const limit = DEFAULT_ROWS_PER_PAGE;
 
 export const Overview = () => {
-  const [isLoading, setIsLoading] = useState(true);
+  const [addPopupOpen, setAddPopupOpen] = useState(false);
+  const [page, setPage] = useState(1);
+  const { appletId } = useParams();
+  const { execute, isLoading, previousValue, value } = useAsync(getAppletSubmissionsApi);
+  const { t } = useTranslation('app');
+  const { data } = value ?? previousValue ?? {};
+  const dashboardTableProps = useMemo(
+    () => mapResponseToSubmissionsTableProps(data ?? ({} as GetAppletSubmissionsResponse)),
+    [data],
+  );
+  const showContent = !isLoading || (isLoading && data);
+
+  const handlePopupClose = (shouldRefetch = false) => {
+    setAddPopupOpen(false);
+
+    if (!!appletId && shouldRefetch) {
+      execute({ appletId, limit });
+    }
+  };
 
   useEffect(() => {
-    // TODO: https://mindlogger.atlassian.net/browse/M2-XXXX
-    // This is just a placeholder until we implement the Overview view
-    window.setTimeout(() => {
-      setIsLoading(false);
-    }, 1000);
-  }, []);
-
-  return <StyledBody>{isLoading ? <Spinner /> : null}</StyledBody>;
+    if (appletId) {
+      execute({ appletId, page, limit });
+    }
+  }, [appletId, execute, page]);
+
+  return (
+    <StyledRoot>
+      {isLoading && <Spinner />}
+
+      {showContent && (
+        <>
+          <QuickStats
+            {...mapResponseToQuickStatProps(data, {
+              onPressAddParticipant: () => {
+                setAddPopupOpen(true);
+              },
+            })}
+          />
+
+          <StyledFlexColumn sx={{ gap: 1.6 }}>
+            <StyledTitleLarge>{t('appletOverview.titleRecentSubmissions')}</StyledTitleLarge>
+
+            <DashboardTable
+              {...dashboardTableProps}
+              handleChangePage={(_, nextPage) => {
+                // DashboardTable passes it's `handleChangePage` prop a 0-based
+                // index value, but accepts its `page` prop as 1-based.
+                setPage(nextPage + 1);
+              }}
+              page={page}
+              rowsPerPage={limit}
+            />
+          </StyledFlexColumn>
+
+          {appletId && (
```

Comment:
I wound up just making the suggested change here (`appletId && addPopupOpen && ...`). I would normally love to ensure that we have properly functioning Modal transitions, but there's just enough complexity here around resetting state, and how the Popup component is actually multiple variable Modal components, and how I'd prefer to tackle this as a more holistic solution that allows us to avoid manually doing the same thing on all other Popups to solve the same problem ... that I think this is best done as a dedicated task on its own.